src/pkg/services/m365/api/groups.go

```diff
@@ -0,0 +1,165 @@
+package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	msgraphgocore "github.com/microsoftgraph/msgraph-sdk-go-core"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/internal/common/tform"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+const (
+	teamsAdditionalDataLabel    = "Team"
+	ResourceProvisioningOptions = "resourceProvisioningOptions"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Teams() Teams {
+	return Teams{c}
+}
+
+// On creation of each Teams team a corrsponding group gets created.
+// The group acts as the protected resource, and all teams data like events,
+// drive and mail messages are owned by that group.
+
+// Teams is an interface-compliant provider of the client.
+type Teams struct {
+	Client
+}
+
+// GetAllTeams retrieves all groups.
+func (c Teams) GetAll(
+	ctx context.Context,
+	errs *fault.Bus,
+) ([]models.Groupable, error) {
+	service, err := c.Service()
+	if err != nil {
+		return nil, err
+	}
+
+	return getGroups(ctx, true, errs, service)
```

Comment:
I'm  playing around with graph api and it looks like the call to `/teams` works just fine as an enumerator, instead of calling `/groups` and filtering down to teams.  

On that note, this file is still conflicted about its identity.  It's called `Groups`, but has a client named `Teams`.  The teams client has a `GetAll` query that calls a `Groups` endpoint.  This all compiles and works out fine, but it's going to be a mess for us to maintain in the long term.  Lets keep it simple for now and have something like this:

```
// in /api/teams.go

func (c Client) Teams() Teams

func (c Teams) GetAll(ctx, errs)
    // should call `/teams`
    
func (c Teams) GetByID(ctx, id, errs)
    // should call `/teams/<id>`
    
func ValidateTeam(...)
```

Simple as that.  Ditch anything groups related until we have clear reason to need it.

Reply:
Yes. That was my first try. But I could not find the GET method for teams in the sdk. Here is the ticket for same - https://github.com/microsoftgraph/msgraph-sdk-go/issues/539. 
Till then I am try to use the raw API.

Reply:
With latest msgraph SDK it works fine. Updated the code.

Reply:
Will require https://github.com/alcionai/corso/pull/3879 to be merged.